packages/astro/src/@types/astro.ts

```diff
@@ -72,21 +72,69 @@ export interface BuildConfig {
 }
 
 /**
- * Astro.* available in all components
- * Docs: https://docs.astro.build/reference/api-reference/#astro-global
+ * Astro global available in all contexts in .astro files
+ *
+ * [Astro documentation](https://docs.astro.build/reference/api-reference/#astro-global)
  */
 export interface AstroGlobal extends AstroGlobalPartial {
-	/** get the current canonical URL */
+	/** Canonical URL of the current page. If the [site](https://docs.astro.build/en/reference/configuration-reference/#site) config option is set, its origin will be the origin of this URL.
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/reference/api-reference/#astrocanonicalurl)
+	 */
 	canonicalURL: URL;
 	/** get page params (dynamic pages only) */
 	params: Params;
-	/** set props for this astro component (along with default values) */
+	/** List of props passed to this component
+	 *
+	 * A common way to get specific props is through destructuring, ex:
+	 * ```javascript
+	 *  const { name } = Astro.props
+	 * ```
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/core-concepts/astro-components/#component-props)
+	 */
 	props: Record<string, number | string | any>;
-	/** get information about this page */
+	/** Information about the current request. This is a standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) object
+	 *
+	 * For example, to get an URL object of the current URL, you can use:
+	 * ```javascript
+	 *  const url = new URL(Astro.request.url);
+	 * ```
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/reference/api-reference/#astrorequest)
+	 */
 	request: Request;
 	/** see if slots are used */
 	slots: Record<string, true | undefined> & {
+		/**
+		 * Check whether content for this slot name exists
+		 *
+		 * Example usage:
+		 * ```typescript
+		 *	if (Astro.slots.has('default')) {
+		 *		// Do something...
+		 *	}
+		 * ```
+		 *
+		 * [Astro documentation](https://docs.astro.build/en/reference/api-reference/#astroslots)
+		 */
 		has(slotName: string): boolean;
```

Comment:
does this mean that you can't have a slot named `has`? 

`slots: Record<string, true | undefined> & { has(slotName: string: boolean; }`

Reply:
Yes, we'll currently throw an error if your slot is named `has` or `get`.